pkg/controllers/flannelmigration/network_migrator.go

```diff
@@ -0,0 +1,191 @@
+	return err
+}
+
+// Drain node, remove Flannel and setup Calico network for a node.
+func (m *networkMigrator) setupCalicoNetworkForNode(node *v1.Node) error {
+	log.Infof("Setting node label to disable Flannel daemonset pod on %s.", node.Name)
+	// Set node label so no Flannel pod can be scheduled on this node.
+	// Flannel pod currently running on the node starts to be evicted as the side effect.
+	n := k8snode(node.Name)
+	err := n.addNodeLabels(m.k8sClientset, nodeNetworkUnknown)
+	if err != nil {
+		log.WithError(err).Errorf("Error adding node label to disable Flannel network for node %s.", node.Name)
+		return err
+	}
+
+	// Cordon and Drain node. Make sure no pod (except daemonset pod or pod with nodeName selector) can run on this node.
+	// TODO
+
+	// Wait for flannel pod to disappear before we proceed, otherwise it may reinstall Flannel network.
+	//n.waitPodsDisappearForNode(m.k8sClientset, 1*time.Second, 2*time.Minute, func(pod *v1.Pod) bool {
+	//	return true
+	//})
+
+	log.Infof("Removing flannel tunnel device/routes on %s.", node.Name)
+	// Remove Flannel network from node.
+	// Note Flannel vxlan tunnel device (flannel.1) is created by Flannel daemonset pod (not Flannel CNI)
+	// Therefor if Flannel daemonset pod can not run on this node, the tunnel device will not be recreated
+	// after we delete it.
+	err = m.removeFlannelNetworkAndInstallDummyCalicoCNI(node)
+	if err != nil {
+		log.WithError(err).Errorf("failed to remove flannel network on node %s", node.Name)
+		return err
+	}
+
+	log.Infof("Deleting non-host-networked pods on %s.", node.Name)
+	// Delete all pods on the node which is not host networked.
+	err = n.deletePodsForNode(m.k8sClientset, func(pod *v1.Pod) bool {
+		return pod.Spec.HostNetwork == false
+	})
+	if err != nil {
+		log.WithError(err).Errorf("failed to delete non-host-networked pods on node %s", node.Name)
+		return err
+	}
+
+	// Set node label so that Calico daemonset pod start to run on this node.
+	// This will install Calico CNI configuration file.
+	// It will take the preference over Flannel CNI config or Canal CNI config.
+	log.Infof("Setting node lable to enable Calico daemonset pod on %s.", node.Name)
+	err = n.addNodeLabels(m.k8sClientset, nodeNetworkCalico)
+	if err != nil {
+		log.WithError(err).Errorf("Error adding node label to enable Calico network for node %s.", node.Name)
+		return err
+	}
+
+	// Calico daemonset pod should start running now.
+	// TODO do we need to wait for calico node pod ready?
+
+	// Uncordon node.
+	//TODO
+
+	log.Infof("Calico networking is running on %s.", node.Name)
+	return nil
+}
+
+// MigrateNodes setup Calico network for array of nodes.
+func (m *networkMigrator) MigrateNodes(nodes []*v1.Node) error {
+	log.Infof("Start network migration process for %d nodes.", len(nodes))
+	for _, node := range nodes {
+		_, err := getNodeLabelValue(m.k8sClientset, node, "node-role.kubernetes.io/master")
+		if err == nil {
+			// TODO Skip master node for now.
```

Comment:
I think it should be safe to do master nodes so long as we don't evict critical pods like the apiserver. Those should be running as static pods in most cases anyway so it's probably OK to just treat them as normal nodes.

Reply:
Sounds good. apiserver is running as a host networked pod so I think we are fine. I will add code so that we migrate master node at later stage

Reply:
Done.